src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -573,7 +570,7 @@ private void finishImportList() {
                     final String nextGroup = getNextImportGroup(currentGroupNumber + 1);
                     if (importGroup.equals(nextGroup)) {
                         if (separateLineBetweenGroups
-                                && getCountOfEmptyLinesBefore(importObject.getLineNumber()) == 0) {
+                                && getCountOfEmptyLinesBefore(importObject.getLineNumber()) != 1) {
```

Comment:
The [documentation](https://checkstyle.org/config_imports.html#CustomImportOrder_Examples) says

>groups are separated by, at least, one blank line

after this change, this becomes 'exactly one blank line'. Please, update the documentation or revert this change. I'd rather to revert. This check is for ordering of imports, not for whitespace validation.

@romani @rnveach please comment.

Reply:
I agree with reverting. Issue is just about not validating empty lines between package and first import.

Reply:
@pbludov @rnveach Please do not miss the context. The original impl logic introduced in #3551 is 
1. check if empty line count before current import <= 1
```
if (getCountOfEmptyLinesBefore(importObject.getLineNumber()) > 1) {
    log(importObject.getLineNumber(), MSG_LINE_SEPARATOR, fullImportIdent);
}
```
2. then when validation empty line betwwen import groups, check if there's line before current import, ie. >=1
```
...
getCountOfEmptyLinesBefore(importObject.getLineNumber()) == 0
...
```

Then if both match, the above logic is identical with `getCountOfEmptyLinesBefore(importObject.getLineNumber()) == 1` but with side effect.

Reply:
I'd prefer updating the documentation to reflect the change.

Reply:
The comment with the PR
```
restore the previous behavior of CustomImportOrder before #3551
```
sounds misleading, let's change it to a better one :smile:

Reply:
OK, let's fix the documentation with a separate PR.

Reply:
@gmshake , please update related issue description to demand update, in other case it will be forgotten.

Reply:
> I'd prefer updating the documentation to reflect the change.

please update documentation to keep it with sync with new behavior.

Reply:
doc issue - https://github.com/checkstyle/checkstyle/issues/7080